File: android/android_mvvm_live_kotlin/app/src/main/java/com/phamnhuvu/android_mvvm_live_kotlin/Injector.kt
package com.phamnhuvu.android_mvvm_live_kotlin

import com.phamnhuvu.android_mvvm_live_kotlin.data.repositories.DemoRepository

class Injector private constructor() {

  lateinit var demoRepository: DemoRepository

  companion object {
    private val mInstance = Injector()

    val instance: Injector
      get() = mInstance
  }

}

File: react_native/react_native_mobx/src/Injector.js
import DemoRepository from "./data/repositories/DemoRepository";
import AppStoreProvider from "./providers/AppStoreProvider";


class Injector {



  // constructor(appStoreProvider) {
  //   this._appStoreProvider = appStoreProvider;
  // }

  get appStoreProvider() {
    return this._appStoreProvider;
  }

  set appStoreProvider(value) {
    this._appStoreProvider = value;
  }

  get demoRepository() {
    return this._demoRepository;
  }

  set demoRepository(value) {
    this._demoRepository = value;
  }
}

const injector = new Injector();
injector.appStoreProvider = new AppStoreProvider();
injector.demoRepository = new DemoRepository();

export default injector;